### PR TITLE
Add email verification

### PR DESCRIPTION
**Pull request by rowannekabalan**
## What does this change?

This updates the developer sign up flow to include an email verification step. Details can be found [here.](https://docs.google.com/document/d/1EThtYQigLa24xoaE4Niec-3iJs5xpDxLDwT5OzjhSak/edit)

## How to test

Deploy to CODE and enable emails in config - you should be able to test the whole sign up flow end-to-end.

## How can we measure success?

People with fake email addresses can't obtain API keys any more.

## Have we considered potential risks?

This is a medium risk change. It introduces potentially breaking changes to the sign up logic, and includes storage changes, UI changes, and more. It has been tested end-to-end but we should keep and eye on it after it goes out.

## Images

In document attached above.

### CODE REVIEW

File: app/controllers/DeveloperForm.scala [hunk 1]
```diff
@@ -27,16 +29,21 @@ class DeveloperForm(override val controllerComponents: ControllerComponents, dyn
...
 
     def handleValidForm(formData: DeveloperCreateKeyFormData): Future[Result] = {
-      logic.createUser(formData) flatMap { consumerKey =>
-        awsEmail.sendEmailNewKey(formData.email, consumerKey) map {
+      logic.createUser(formData) flatMap { bonoboUser =>
+        awsEmail.sendEmailVerification(formData.email, verificationUrl(bonoboUser)) map {
           result => Redirect(routes.DeveloperForm.complete)
         } recover {
-          case _ => Redirect(routes.DeveloperForm.complete).flashing("error" -> s"We were unable to send the email with the new key to ${formData.email}.  Please contact content.delivery@theguardian.com for further instructions.")
+          case _ => Redirect(routes.DeveloperForm.complete).flashing("error" -> s"We were unable to send verification email to ${formData.email}.  Please contact content.delivery@theguardian.com for further instructions.")
```
Comment (JustinPinner) [inline app/controllers/DeveloperForm.scala:46]: Is `content.delivery` still a valid address?

File: conf/application.conf [hunk 2]
```diff
@@ -56,4 +56,6 @@ recaptcha.privateKey="changeme"
 play.filters.headers {
   contentSecurityPolicy = "default-src 'self';"
   contentSecurityPolicy = ${play.filters.headers.contentSecurityPolicy}" script-src 'self' 'unsafe-inline' google.com/recaptcha;"
-}
\ No newline at end of file
+}
+
+bonoboUrl = "http://localhost:9000"
```
Comment (JustinPinner) [inline conf/application.conf:61]: Is this needed still?

**Review by JustinPinner — APPROVED** [review]
Just a couple of small questions but looking good to go I think (esp. as it has been throughly tested) 👍

**Review by JustinPinner — APPROVED** [review]
Looking even better now! 👍

File: conf/application.conf (hunk 2 above)
Comment (rowannekabalan) [reply conf/application.conf:61]: I think it's fine to leave it there as this file is dev conf and also referenced as a template of what the config file should look like

**Pull request merged**